docs/rules/padding-line-between-statements.md

```diff
@@ -0,0 +1,229 @@
+You can supply any number of the configuration. If an statement pair matches some configurations, it chooses the last matched configuration.
+
+```json
+{
+    "padding-line-between-statements": ["error", [
+        { "blankline": LINEBREAK_TYPE, "prev": STATEMENT_TYPE, "next": STATEMENT_TYPE },
+        { "blankline": LINEBREAK_TYPE, "prev": STATEMENT_TYPE, "next": STATEMENT_TYPE },
+        { "blankline": LINEBREAK_TYPE, "prev": STATEMENT_TYPE, "next": STATEMENT_TYPE },
+        { "blankline": LINEBREAK_TYPE, "prev": STATEMENT_TYPE, "next": STATEMENT_TYPE },
+        ...
+    ]]
+}
+```
+
+- `LINEBREAK_TYPE` is one of them.
+    - `"any"` just ignores the statement pair.
+    - `"never"` disallows blank lines.
+    - `"always"` requires one or more blank lines. Note it does not count lines that comments exist as blank lines.
+
+- `STATEMENT_TYPE` is one of them, or an array of them.
+    - `"*"` is wildcard. This matches any statements.
+    - `"block"` is lonely blocks.
+    - `"block-like"` is block like statements. This matches statements that the last token is the closing brace of blocks; e.g. `{ }`, `if (a) { }`, and `while (a) { }`.
+    - `"break"` is `break` statements.
+    - `"case"` is `case` labels.
+    - `"cjs-export"` is `export` statements of CommonJS; e.g. `module.exports = 0`, `module.exports.foo = 1`, and `exports.foo = 2`. This is the special cases of assignment.
+    - `"cjs-import"` is `import` statements of CommonJS; e.g. `const foo = require("foo")`. This is the special cases of variable declarations.
+    - `"class"` is `class` declarations.
+    - `"const"` is `const` variable declarations.
+    - `"continue"` is `continue` statements.
+    - `"debugger"` is `debugger` statements.
+    - `"default"` is `default` labels.
+    - `"directive"` is directive prologues. This matches directives; e.g. `"use strict"`.
+    - `"do"` is `do-while` statements. This matches all statements that the first token is `do` keyword.
+    - `"empty"` is empty statements.
+    - `"export"` is `export` declarations.
+    - `"expression"` is expression statements.
+    - `"for"` is `for` loop families. This matches all statements that the first token is `for` keyword.
+    - `"function"` is function declarations.
+    - `"if"` is `if` statements.
+    - `"import"` is `import` declarations.
+    - `"let"` is `let` variable declarations.
+    - `"multiline-block-like"` is block like statements. This is the same as `block-like` type, but only the block is multiline.
+    - `"return"` is `return` statements.
+    - `"switch"` is `switch` statements.
+    - `"throw"` is `throw` statements.
+    - `"try"` is `try` statements.
+    - `"var"` is `var` variable declarations.
+    - `"while"` is `while` loop statements.
+    - `"with"` is `with` statements.
+
+## Examples
+
+It configures as [newline-before-return].
```

Comment:
Nitpick: I think it would be better to explain this a bit more. For example:

> This configuration would require blank lines before all `return` statements, like the `newline-before-return` rule.